### PR TITLE
Added new file publisher plugin location

### PR DESCRIPTION
**Pull request by thomastaylor312**
Summary of changes:
- Adds file plugin to catalog

Testing done:
- None needed

@intelsdi-x/snap-maintainers

### CODE REVIEW

**Comment by andrzej-k** [comment]
LGTM with 2 observations:

- At some point plugins were added based on their type - first collector, then processor and publisher plugins. That doesn't seem to be the case anymore, and they are just added alphabetically. We should either re-sort this list or create sections per plugin type. 

- File publisher plugin still needs some updates in Readme; sections "Documentation" and "Examples" are TODO

**Comment by thomastaylor312** [comment]
@andrzej-k I was planning on updating the README today. I'll update this PR once I am done

**Comment by thomastaylor312** [comment]
README updated in [this](https://github.com/intelsdi-x/snap-plugin-publisher-file/pull/6) PR. As for the catalog sectioning, should we open a separate issue @andrzej-k?

File: docs/PLUGIN_CATALOG.md [hunk 1]
```diff
@@ -15,6 +15,7 @@ This is the master catalog of plugins for snap. The plugins in this list may be
 | Etcd | Collector | Collects metrics from Etcd's `/metrics` endpoint. | [snap-plugin-collector-etcd](https://github.com/intelsdi-x/snap-plugin-collector-etcd) |
 | Ethtool | Collector | Collect from ethtool stats & registry dump |[snap-plugin-collector-ethtool](https://github.com/intelsdi-x/snap-plugin-collector-ethtool) | [Linux](https://s3-us-west-1.amazonaws.com/snap-plugins-linux-latest/snap-plugin-collector-ethtool)
 | Facter | Collector | Collects from Facter | [snap-plugin-collector-facter](https://github.com/intelsdi-x/snap-plugin-collector-facter) | [Linux](https://s3-us-west-1.amazonaws.com/snap-plugins-linux-latest/snap-plugin-collector-facter) &#124; [Darwin](https://s3-us-west-1.amazonaws.com/snap-plugins-darwin-latest/snap-plugin-collector-facter)
+| File | Publisher | Publishes snap metrics to a file | [snap-plugin-publisher-file](https://github.com/intelsdi-x/snap-plugin-publisher-file) |
```
Comment (IzabellaRaulin) [inline docs/PLUGIN_CATALOG.md]: Consider mention in description that this this plugin publishes in JSON format.

**Comment by andrzej-k** [comment]
@thomastaylor312: yes, I've opened one: #1067 
LGTM

File: docs/PLUGIN_CATALOG.md (hunk 1 above)
Comment (thomastaylor312) [inline docs/PLUGIN_CATALOG.md]: Ok, I'll add that and merge.

**Pull request merged**